include/fractionalintegralmodel.h

```diff
@@ -0,0 +1,95 @@
+#ifndef GAUSS2D_FIT_FRACTIONALINTEGRALMODEL_H
+#define GAUSS2D_FIT_FRACTIONALINTEGRALMODEL_H
+
+#include <map>
+#include <memory>
+#include <vector>
+
+#include "component.h"
+#include "param_defs.h"
+#include "param_filter.h"
+#include "parameters.h"
+#include "integralmodel.h"
+
+namespace gauss2d
+{
+namespace fit
+{
+
+class FractionalGaussianIntegral;
+
+class FractionalIntegralModel : public IntegralModel {
+public:
+    typedef std::map<
+        std::reference_wrapper<const Channel>,
+        std::shared_ptr<ProperFractionParameter>,
+        std::less<const Channel>
+    > Data;
+
+private:
+    Data _data = {};
+    const FractionalIntegralModel * _find_parent(const IntegralModel * model);
+    const std::shared_ptr<const IntegralModel> _model;
+    const FractionalIntegralModel * _parent;
+    static inline std::map<const FractionalIntegralModel*, const IntegralModel*> _registry = {};
```

Comment:
I managed to change all of the raw pointers to `weak_ptr` or `reference_wrapper` as appropriate, but there is a bit of a cost. `reference_wrapper` requires a `<` operator, which I defined for `IntegralModel`. Also, I made the constructor static like with `Channel`. Otherwise, it can't add a `weak_ptr` of `self` to the registry unless it uses `enable_shared_from_this` (which I wasn't exactly opposed to, but considering that static constructors can be bound to `init` with pybind11, that seems like the better option).